ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -194,6 +197,53 @@ def _llmobs_set_meta_tags_from_chain(
             except TypeError:
                 log.warning("Failed to serialize chain output data to JSON")
 
+    def _llmobs_set_meta_tags_from_similarity_search(
+        self,
+        span: Span,
+        input_query: str,
+        output_documents: Union[List[Any], None],
+        error: bool = False,
+        is_workflow: bool = False,
+    ) -> None:
+        span.set_tag_str(SPAN_KIND, "workflow" if is_workflow else "retrieval")
+        span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
+        span.set_tag_str(MODEL_PROVIDER, span.get_tag(PROVIDER) or "")
+
+        input_tag_key = INPUT_VALUE if is_workflow else INPUT_MESSAGES
+        output_tag_key = OUTPUT_VALUE if is_workflow else OUTPUT_DOCUMENTS
+
+        if input_query is not None:
+            try:
+                formatted_inputs = self.format_io(input_query)
+                if isinstance(formatted_inputs, str):
+                    span.set_tag_str(input_tag_key, formatted_inputs)
+                else:
+                    span.set_tag_str(input_tag_key, json.dumps(self.format_io(input_query)))
+            except TypeError:
+                log.warning("Failed to serialize similarity input query to JSON")
+        if error:
+            span.set_tag_str(output_tag_key, "")
+        elif isinstance(output_documents, list):
+            if is_workflow:
+                span.set_tag_str(output_tag_key, json.dumps(output_documents))
+            else:
+                try:
+                    documents = [
+                        Document(
+                            id=d.id,
+                            text=d.page_content,
+                            name=d.metadata["source"],
```

Comment:
Is there documentation anywhere where we can be sure that Langchain documents will always follow this schema?

Reply:
I believe source is the one not mentioned in the schema https://api.python.langchain.com/en/latest/documents/langchain_core.documents.base.Document.html#langchain_core.documents.base.Document
- Id is there even though it's optional
- metadata is said to be an arbitrary list of tags

I'll update this to only pick up the id and page_content
But where should we be putting the all the metadata ? I noticed there's a Metadata slot but it's more about the whole LLM operation metadata and not each document from the output.

Reply:
we can leave out metadata until we add support for storing it in our Documents data class in the backend. It seems like ID is optional so let's default to an empty string if `document.id` is None.